### PR TITLE
Fix panic in controller framework

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Fixes https://github.com/kubernetes/kubernetes/issues/25594

/cc @kubernetes/sig-api-machinery

### CODE REVIEW

**Comment by hongchaodeng** [comment]
In what case will queue be nil?

File: pkg/controller/framework/controller.go [hunk 1]
```diff
@@ -105,7 +105,10 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 
 // Returns true once this controller has completed an initial resource listing
 func (c *Controller) HasSynced() bool {
-	return c.config.Queue.HasSynced()
+	if c.config.Queue != nil {
```
Comment (lavalamp) [inline pkg/controller/framework/controller.go]: Checking for nil here is probably racy, do you need a lock? Actually how is this possible at all, there may be a bug elsewhere if it's possible to get a Controller pointer with a nil Queue?

Comment (hongchaodeng) [inline pkg/controller/framework/controller.go]: I agree. A controller shouldn't have nil Queue. There are many places that use Queue directly and checking here not there would make it more fragmented.

Comment (dims) [inline pkg/controller/framework/controller.go]: This just feels weird :) agreeing with @lavalamp and @hongchaodeng

**Comment by derekwaynecarr** [comment]
It's possible the code that spurred the flake changed in the interim.  Will close this now.

**Pull request closed without merging**